src/parser.cpp

```diff
@@ -1266,7 +1266,11 @@ DataBuffer* BotscriptParser::parseExpression (DataType reqtype, bool fromhere)
 	if (fromhere)
 		m_lexer->skip (-1);
 
-	Expression expr (this, m_lexer, reqtype);
+	Expression expr (this, m_lexer, TYPE_ToBeDecided);
+	ASSERT_NE(expr.getResult()->valueType(), TYPE_ToBeDecided)
+	if (expr.getResult()->valueType() != reqtype) {
+		error("Incompatible data type");
```

Comment:
error: crashbot.botc:64:near 51: Incompatible data type
`if (ChatSectionExistsInChatLump("WinStrings")) {`

Reply:
Try `botc_defs.bts` from this branch. 1060356 requires `if` "argument" to be boolean. (This is what Java does and I think such behavior is more friendly to Doom modders)
When an actual int returning function is used as `if` argument, you need to rewrite it like `if (func() != 0)`.

Reply:
Something I noticed, although I don't know if you would consider it a bug. The compiler complains if the logical operator is used in an int (let's suppose it is named $local1) like `$local1 = !($local1)`. Still converting the source code for the new defs, so far no real issues.

Reply:
It compiled and works with the Roam() fix, here is the script I've used <http://sprunge.us/v4r7jN>.